staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/values.go

```diff
@@ -35,6 +35,12 @@ import (
 // The provided schema is expected to exactly match the fields we expose to CEL. I.e.
 // for resource roots, type meta and object meta schemas must be present.
 func UnstructuredToVal(unstructured interface{}, schema *structuralschema.Structural) ref.Val {
+	if unstructured == nil {
+		if schema.Nullable {
+			return types.NullValue
+		}
+		return types.NewErr("invalid data, got null for schema with nullable=false")
```

Comment:
this seems fine as a backstop, but doesn't PruneNonNullableNullsWithoutDefaults / Default+isNonNullableNull ensure we won't actually encounter this in real life?

If I submit an explicit null value for a non-nullable item, I think it either gets pruned (if no default is defined) or defaulted (if a default is defined)

Reply:
Yep, these were intended as backstops. So long as OpenAPI rejects the data these lines are unreachable. So if any of these ever surface, it means we've got a bug.